lib/web_ui/lib/pointer.dart

```diff
@@ -153,7 +153,23 @@ class PointerData {
   }
 }
 
+/// A function used to notify the engine about how a [PointerDataPacket] was used in the framework.
+///
+/// For example, can be used to [preventDefault] the source events on the web.
+typedef _PointerDataPacketAcknowledgeFn = void Function({
+  bool? preventDefault,
+});
+
+// The default _PointerDataPacketAcknowledgeFn, which is a noop.
+void _defaultPointerDataPacketAcknowledgeFn ({
+  bool? preventDefault,
+}) {}
+
 class PointerDataPacket {
-  const PointerDataPacket({this.data = const <PointerData>[]});
+  const PointerDataPacket({
+    this.data = const <PointerData>[],
+    this.acknowledge = _defaultPointerDataPacketAcknowledgeFn,
+  });
   final List<PointerData> data;
+  final _PointerDataPacketAcknowledgeFn acknowledge;
```

Comment:
This should trigger the `library_private_types_in_public_api` lint (surprised it didn't). Nevertheless:

> Avoid using private types in public APIs.

https://dart.dev/tools/linter-rules/library_private_types_in_public_api

Reply:
I saw this (in the framework repo maybe?), and was surprised it didn't trigger here. I think I'm going to expand the definition to `void Function(...blah` so I don't expose a new type?